hs_core/hydro_realtime_signal_processor.py

```diff
@@ -59,15 +59,57 @@ def handle_save(self, sender, instance, **kwargs):
         elif isinstance(instance, ResourceAccess):
             # automatically a BaseResource; just call the routine on it. 
             newinstance = instance.resource
-            newsender = BaseResource
-            self.handle_save(newsender, newinstance)
+            self.handle_save(BaseResource, newinstance)
 
+        elif isinstance(instance, CoreMetaData): 
+            try: 
+                newinstance = BaseResource.objects.get(metadata=instance)
+                self.handle_save(BaseResource, newinstance)
+            except Exception: 
+                pass
+        
+        elif isinstance(instance, AbstractMetaDataElement):
+            try: 
+                # resolve the BaseResource corresponding to the metadata element. 
+                # fields used here are the union of all fields in the metadata element
+                # some of these are deprecated, but in the interest of time, 
+                # I included them now and will delete later. 
+                newinstance = BaseResource.objects.get(Q(metadata__title=instance) |
```

Comment:
This can be collapsed to `newinstance = instance.metadata.resource` and will future proof for new/removed metadata elements.  It should also work for aggregation metadata... I'm pretty sure that's a true statement.

Reply:
Changed. I need to check carefully for aggregated metadata.